packages/backend-for-frontend/src/services/agreementService.ts

```diff
@@ -112,6 +110,51 @@ export function agreementServiceBuilder(
             showOnlyUpgradeable,
             eservicesIds,
             consumersIds,
+            states,
```

Comment:
See comments above: here we should fix the consumer to the requester and leave the producer filter customizable by the front-end. Something like:

```suggestion
        await agreementProcessClient.getAgreements({
          queries: {
            states,
            producersIds,
            consumersIds: [ctx.authData.organizationId],
          },
```